src/fsharp/ConstraintSolver.fs

```diff
@@ -835,10 +835,26 @@ and SolveAnonInfoEqualsAnonInfo (csenv: ConstraintSolverEnv) m2 (anonInfo1: Anon
     (match anonInfo1.Assembly, anonInfo2.Assembly with 
         | ccu1, ccu2 -> if not (ccuEq ccu1 ccu2) then ErrorD (ConstraintSolverError(FSComp.SR.tcAnonRecdCcuMismatch(ccu1.AssemblyName, ccu2.AssemblyName), csenv.m,m2)) else ResultD ()
         ) ++ (fun () -> 
+
     if not (anonInfo1.SortedNames = anonInfo2.SortedNames) then 
-        let namesText1 = sprintf "%A" (Array.toList anonInfo1.SortedNames)
-        let namesText2 = sprintf "%A" (Array.toList anonInfo2.SortedNames)
-        ErrorD (ConstraintSolverError(FSComp.SR.tcAnonRecdFieldNameMismatch(namesText1, namesText2), csenv.m,m2)) 
+        let (|Subset|Superset|Overlap|CompletelyDifferent|) (first, second) =
+            let first = Set first
+            let second = Set second
+            let secondOnly = (second - first) |> Set.toList
+            let firstOnly = (first - second) |> Set.toList
+            if second.IsSubsetOf first then Subset firstOnly
+            elif second.IsSupersetOf first then Superset secondOnly
+            elif Set.intersect first second <> Set.empty then Overlap(firstOnly, secondOnly)
+            else CompletelyDifferent(Seq.toList first)
```

Comment:
Typically the style in the compiler is to have the body in a new indented line, so I think that would be preferred here.

Note: there are many styles in this codebase :(

Reply:
Do you mean that each expression after the `then` should be on its own line? Happy to do that - but even `ConstraintSolver.fs` has multiple styles. Is there any rule here?

Reply:
Yes, that's what I mean. There's no rules here really. When a file has consistent style we try to keep it that way but when there's new code being added and no real precedent we tend to place expressions on new lines since that is the most common across the whole codebase.

Reply:
Done.